lib/smack/SmackRep.cpp

```diff
@@ -833,6 +837,9 @@ const Expr* SmackRep::bop(const llvm::BinaryOperator* BO) {
 
 const Expr* SmackRep::bop(unsigned opcode, const llvm::Value* lhs, const llvm::Value* rhs, const llvm::Type* t) {
   std::string fn = Naming::INSTRUCTION_TABLE.at(opcode);
+  if (fn == "$fadd" || fn == "$fsub" || fn == "$fmul" || fn == "$fdiv") {
```

Comment:
Can't you use `opcode` here instead of `fn`?

Reply:
Yes, this is a much better idea.